4-perftools/test/run

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+#
+# The 'run' script performs simple tests that verifies usability
+# of tools, packaged in perftools image.
+#
+# IMAGE_NAME specifies a name of the candidate image used for testing.
+# The image has to be available before this script is executed.
+#
+# DEBUG environment variable, if not empty, makes 'run' to log every step
+# of testing.
+#
+
+if [ "$DEBUG" != "" ]; then
+  set -x
+fi
+
+IMAGE_NAME=${IMAGE_NAME:-rhscl/devtoolset-4-perftools-rhel7}
+
+function info () {
+    echo -e "\e[1m[INFO] $@\e[0m"
+}
+
+function pass () {
+    echo -e "\e[1;32m[PASS] $@\e[0m"
+}
+
+function error () {
+    echo -e "\e[1;31m[ERROR] $@\e[0m"
+}
+
+function check_result() {
+    local label="$1"
+    local result="$2"
+    local expected="$3"
+
+    if [[ "$result" = "$expected" ]]; then
+        pass "$label: PASS"
+    else
+        error "$label: FAIL ($result)"
+        RESULT=1
+    fi
+}
+
+function test_docker_run_usage () {
+  info "Testing 'docker run' usage ..."
+
+  sed -i "s/RELEASE/$(docker run --rm $IMAGE_NAME uname -r)/" expected-usage
+
+  docker run --rm $IMAGE_NAME > $TMPDIR/actual-usage
+  check_result "Exit code is zero" $? 0
+
+  diff expected-usage $TMPDIR/actual-usage &> usage-diff
+  check_result "Usage info matches the expected text" $? 0
+}
+
+function test_sanity_valgrind_usage () {
+  info "Testing 'valgrind --version' usage ..."
+
+  docker run --rm $IMAGE_NAME valgrind --version &> $TMPDIR/actual-valgrind-version
+  check_result "Exit code is zero" $? 0
+
+  grep 'valgrind-3.11.0' $TMPDIR/actual-valgrind-version &> /dev/null
+  check_result "Output contains valgrind version" $? 0
+}
+
+function test_sanity_valgrind_run () {
+  info "Testing 'valgrind /foo/ls' usage ..."
+
+  docker run --rm -v /bin:/foo:ro $IMAGE_NAME valgrind /foo/ls &> $TMPDIR/actual-valgrind-run
+  check_result "Exit code is zero" $? 0
+
+  grep 'ERROR SUMMARY: 0 errors from 0 contexts' $TMPDIR/actual-valgrind-run &> /dev/null
+  check_result "Output contains valgrind summary" $? 0
+}
+
+function test_sanity_eu_objdump_run () {
+  info "Testing 'eu-objdump /foo/ls' usage ..."
+
+  docker run --rm -v /bin:/foo:ro $IMAGE_NAME eu-objdump -d /foo/ls &> $TMPDIR/actual-objdump
+  check_result "Exit code is zero" $? 0
+
+  grep 'Disassembly of section' $TMPDIR/actual-objdump &> /dev/null && grep 'movzbl' $TMPDIR/actual-objdump &> /dev/null
+  check_result "Output contains disassembled code" $? 0
+}
+
+function test_sanity_ocount_run () {
+  info "Testing 'ocount ls' usage ..."
+
+  docker run --rm $IMAGE_NAME ocount ls &> $TMPDIR/actual-ocount
+  check_result "Exit code is zero" $? 1
+
+  grep "Your kernel's Performance Events Subsystem does not support your processor type." $TMPDIR/actual-ocount &> /dev/null
+  check_result "Output contains ocount's complaints" $? 0
```

Comment:
This test `test_sanity_ocount_run` does not work for me, aren't there the exit codes swapped? `ocount ls` should exit with 0 and grep with 1, shouldn't it?